digital_printing/digital_printing/doctype/print_order/print_order.js

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) 2023, ParaLogic and contributors
+// For license information, please see license.txt
+
+frappe.provide("erpnext.digital_printing");
+
+erpnext.digital_printing.PrintOrder = class PrintOrder extends frappe.ui.form.Controller {
+	setup() {
+		this.setup_queries();
+	}
+
+	refresh() {
+		erpnext.hide_company();
+	}
+
+	setup_queries() {
+		this.frm.set_query("fabric_item", () => {
+			let filters = {
+				'print_item_type': 'Fabric'
+			}
+			if (this.frm.doc.is_fabric_provided_by_customer) {
+				filters.customer = this.frm.doc.customer;
+			}
+			return { filters: filters }
+		});
+
+		this.frm.set_query("process_item", () => {
+			return {
+				filters: { print_item_type: 'Print Process' }
+			}
+		});
+	}
+
+	on_upload_complete() {
+		return this.get_items_from_attachments();
+	}
+
+	default_gap() {
+		this.override_default_value_in_items('design_gap');
+	}
+
+	default_qty() {
+		this.override_default_value_in_items('qty');
+	}
+
+	default_uom() {
+		this.override_default_value_in_items('uom');
+		if (this.frm.doc.default_uom == "Panel") {
+			this.frm.set_value("default_qty_type", "Print Qty");
+		} else {
+			this.frm.set_value("default_length_uom", this.frm.doc.default_uom);
+		}
+	}
+
+	default_qty_type() {
+		this.override_default_value_in_items('qty_type');
+	}
+
+	default_wastage() {
+		this.override_default_value_in_items('per_wastage');
+	}
+
+	default_length_uom() {
+		this.override_default_value_in_items('length_uom');
+	}
+
+	items_add(doc, cdt, cdn) {
+		this.set_default_values_in_item(cdt, cdn);
+	}
+
+	before_items_remove(doc, cdt, cdn) {
+		let row = frappe.get_doc(cdt, cdn);
+		let file_name = this.frm.attachments.get_file_id_from_file_url(row.design_image);
+		this.frm.attachments.remove_attachment(file_name);
+		this.calculate_totals();
+	}
+
+	design_image(doc, cdt, cdn) {
+		var me = this;
+		let row = frappe.get_doc(cdt, cdn);
+
+		return frappe.call({
+			method: "get_image_details",
+			args: {
+				image_url: row.design_image
+			},
+			doc: me.frm.doc,
+			callback: function(r) {
+				if (!r.exc && r.message) {
+					for (const [key, value] of Object.entries(r.message)) {
+						frappe.model.set_value(cdt, cdn, key, value);
```

Comment:
why not just `return frappe.model.set_value(cdt, cdn, r.message)` instead of looping